plugins/importexport/onix30/filter/MonographONIX30XmlFilter.inc.php

```diff
@@ -515,6 +515,12 @@ function createProductNode($doc, $submission, $publicationFormat) {
 		$websiteNode->appendChild($this->_buildTextNode($doc, 'WebsiteRole', '18')); // 18 -> Publisher's B2C website
 		$websiteNode->appendChild($this->_buildTextNode($doc, 'WebsiteLink', $request->url($context->getPath())));
 
+		$websiteNode = $doc->createElementNS($deployment->getNamespace(), 'Website');
+		$publisherNode->appendChild($websiteNode);
+
+		$websiteNode->appendChild($this->_buildTextNode($doc, 'WebsiteRole', '29')); // 29 -> Web page for full content
+		$websiteNode->appendChild($this->_buildTextNode($doc, 'WebsiteLink', $request->url($context->getPath(), 'catalog', 'book', $publication->getData('submissionId'))));
```

Comment:
The submission ID will always be available for the public-facing URL, but some editors will also use a custom URL slug; when generating a URL these should be preferred if they're available. You can get these with: `$publication->getData('urlPath')`